R/op-bitcast_convert.R

```diff
@@ -0,0 +1,94 @@
+#' @include op.R hlo.R
+NULL
+
+OpBitcastConvert <- new_Op("OpBitcastConvert", "bitcast_convert")
+
+infer_types_bitcast_convert <- function(
+  operand,
+  dtype
+) {
+  stopifnot(inherits(operand@type, TensorType))
+
+  if (
+    dtype %in% c("i1", "pred") || (inherits(operand@type@dtype, BooleanType))
+  ) {
+    cli_abort("Bitcast conversions from and to booleans are not supported.")
+  }
+
+  stopifnot(
+    dtype %in%
+      c(
+        "i8",
+        "i16",
+        "i32",
+        "i64",
+        "ui8",
+        "ui16",
+        "ui32",
+        "ui64",
+        "f8",
+        "f16",
+        "f32",
+        "f64"
+      )
+  )
+
+  operand_bits <- operand@type@dtype@value
+  operand_dims <- shape(operand)
+
+  output_bits <- as.integer(sub(".*?([0-9]+)$", "\\1", dtype))
+  cst_fct <- output_bits / operand_bits
+
+  if (cst_fct == 1) {
+    result_dims <- operand_dims
+  } else if (cst_fct > 1) {
+    if (operand_dims[[length(operand_dims)]] != cst_fct) {
+      cli_abort(sprintf(
```

Comment:
`cli_abort()` supports things like `cli_abort("blabla {d}")`, so no need for sprintf here